src/main/kotlin/org/rust/lang/core/RustPsiPattern.kt

```diff
@@ -0,0 +1,35 @@
+import com.intellij.patterns.PatternCondition
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiErrorElement
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.RustTokenElementTypes
+
+/**
+ * Rust PSI tree patterns.
+ */
+class RustPsiPattern {
```

Comment:
Although majority of test are integrated, it makes sense to add "unit tests" for patterns specifically, because they are very sensitive to PSI strucutre, that changes oftentimes.